pkg/pxc/app/statefulset/node.go

```diff
@@ -211,6 +225,102 @@ func (c *Node) SidecarContainers(spec *api.PodSpec, secrets string, cr *api.Perc
 	return nil, nil
 }
 
+func (c *Node) LogCollectorContainer(spec *api.LogCollectorSpec, logPsecrets string, logRsecrets string, cr *api.PerconaXtraDBCluster) ([]corev1.Container, error) {
+	res, err := app.CreateResources(spec.Resources)
+	if err != nil {
+		return nil, fmt.Errorf("create resources error: %v", err)
+	}
+
+	logProcEnvs := []corev1.EnvVar{
+		{
+			Name:  "LOG_DATA_DIR",
+			Value: "/var/lib/mysql",
+		},
+		{
+			Name: "POD_NAMESPASE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		},
+	}
+
+	logRotEnvs := []corev1.EnvVar{
+		{
+			Name:  "SERVICE_TYPE",
+			Value: "mysql",
+		},
+		{
+			Name: "MONITOR_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: app.SecretKeySelector(logRsecrets, "monitor"),
+			},
+		},
+	}
+
+	fvar := true
+	logProcContainer := corev1.Container{
+		Name:            "logs",
+		Image:           spec.Image,
+		ImagePullPolicy: spec.ImagePullPolicy,
+		Env:             logProcEnvs,
+		SecurityContext: spec.ContainerSecurityContext,
+		Resources:       res,
+		EnvFrom: []corev1.EnvFromSource{
+			{
+				SecretRef: &corev1.SecretEnvSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: logPsecrets,
+					},
+					Optional: &fvar,
+				},
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      DataVolumeName,
+				MountPath: "/var/lib/mysql",
+			},
+		},
+	}
+
+	logRotContainer := corev1.Container{
+		Name:            "logrotate",
+		Image:           spec.Image,
+		ImagePullPolicy: spec.ImagePullPolicy,
+		Env:             logRotEnvs,
+		SecurityContext: spec.ContainerSecurityContext,
+		Resources:       res,
+		Args: []string{
+			"logrotate",
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      DataVolumeName,
+				MountPath: "/var/lib/mysql",
+			},
+		},
+	}
+
+	if cr.Spec.LogCollector != nil && cr.Spec.LogCollector.Configuration != "" {
+		logProcContainer.VolumeMounts = append(logProcContainer.VolumeMounts, corev1.VolumeMount{
+			Name:      "custom-config",
```

Comment:
can we please rename?
```suggestion
			Name:      "logcollector-config",
```

Reply:
done